frontend/lib/shared/number_pad.dart

```diff
@@ -52,13 +49,11 @@ class NumberPad extends HookWidget {
 
 class NumberPadKey extends HookWidget {
   final String title;
-  final Function(String) onKeyPressed;
-  final VoidCallback onDeletePressed;
+  final ValueNotifier<String> enteredAmount;
```

Comment:
I wonder if a `ValueNotifier` is the best approach. Honestly not sure what would be best.

We could also just have like an `initialValue = 0` and a callback with `valueChanged`. Then the users of NumberPad could just hook into the callback. Whatcha think?

Reply:
It also feels like the thing that is sent back from NumberPad should be a "number" like `double` vs. a `string`